cmake/modules/hwm_v2.cmake

```diff
@@ -22,6 +22,8 @@ if(NOT HWMv2)
   return()
 endif()
 
+set_ifndef(KCONFIG_BINARY_DIR ${CMAKE_BINARY_DIR}/Kconfig)
```

Comment:
I wonder how is this change related to what is put into the commit description ?

Reply:
Afaict, the initial value of `KCONFIG_BINARY_DIR` is usually set by `zephyr_module.cmake`. However, that file is also loaded in unit test mode, so I'm not sure why `hwm_v2.cmake` couldn't be loaded after it, instead of adding this line.

Reply:
> so I'm not sure why `hwm_v2.cmake` couldn't be loaded after it, instead of adding this line.

Good question. I can't see why https://github.com/zephyrproject-rtos/zephyr/pull/64401 added the lines at the current locations instead of following the same structure used in `zephyr_default.cmake`.

Reply:
Reordered `include`s to follow same order as zephyr_default.cmake.